tokens.xml

```diff
@@ -8438,6 +8654,173 @@ Sacrifice this creature: This creature deals 1 damage to target creature or play
             <token>1</token>
             <tablerow>2</tablerow>
         </card>
+        <card>
+            <name>Dungeon of the Mad Mage</name>
+            <text>Yawning Portal — You gain 1 life. (→ Dungeon Level)
+Dungeon Level — Scry 1. (→ Goblin Bazaar or Twisted Caverns)
+Goblin Bazaar — Create a Treasure token. (→ Lost Level)
+Twisted Caverns — Target creature can’t attack until your next turn. (→ Lost Level)
+Lost Level — Scry 2. (→ Runestone Caverns or Muiral’s Graveyard)
+Runestone Caverns — Exile the top two cards of your library. You may play them. (→ Deep Mines)
+Muiral’s Graveyard — Create two 1/1 black Skeleton creature tokens. (→ Deep Mines)
+Deep Mines — Scry 3. (→ Mad Wizard’s Lair)
+Mad Wizard’s Lair — Draw three cards and reveal them. You may cast one of them without paying its mana cost.</text>
+            <prop>
+                <type>Dungeon</type>
+                <maintype>Dungeon</maintype>
+            </prop>
+            <set picURL="https://c1.scryfall.com/file/scryfall-cards/large/front/6/f/6f509dbe-6ec7-4438-ab36-e20be46c9922.jpg?1625106684">AFR</set>
+            <related count="2" exclude="exclude">Skeleton </reverse-related>
```

Comment:
You have mismatched start and end tags here. This breaks the parsing for the rest of the file.